src/mds/PurgeQueue.cc

```diff
@@ -251,6 +262,13 @@ void PurgeQueue::create(Context *fin)
   dout(4) << "creating" << dendl;
   std::lock_guard l(lock);
 
+  if (readonly) {
+    dout(10) << "cannot create: PurgeQueue is readonly" << dendl;
+    if (fin)
+      fin->complete(-EROFS);
+    return;
```

Comment:
this does not make sense for creating new prugequeue. Instead, we should check error in finish context of writing journal head. (at later this function)